tests/unit/test_endpoint_provider.py

```diff
@@ -411,6 +411,14 @@ def test_alerts_in_both_formats_are_recursively_aggregated(self):
 
         self.assertEqual({}, DeepDiff(expected_rules_file, rules_file_dict, ignore_order=True))
 
+    def test_unit_not_in_alert_labels(self):
```

Comment:
I would have expected a couple of more unit tests here, because the behaviours we are concerned with are

- Does `MetricsEndpointProvider` **not add** `juju_unit` to alert rules expressions and labels ?
- Does `MetricsEndpointProvider` **retain** `juju_unit` if the are present in alert rule expressions and labels ?
- Does `MetricsEndpointAggregator` **not add** `juju_unit` to alert rules expressions and labels ?
- Does `MetricsEndpointAggregator` **retain** `juju_unit` if the are present in alert rule expressions and labels ?

Reply:
I'm not sure if this will strip out existing unit tags. @mmanciop What is the expected behavior here?

Reply:
If the alert rule does come with unit tags (which might be needed for some corner cases of the cos-configuration-repository charm), we need to keep that label.

Reply:
A unit test has been added for the case of existing labels